### DISCUSSION TITLE
Runtime Error When Training RAVE Model with Preprocessed Dataset

### DISCUSSION THREAD

**Discussion by azeriff** [acids-ircam/RAVE#280] [Q&A]
Hi everyone,

I've been trying to train a RAVE model in Google Colab using a preprocessed dataset but have encountered a recurring runtime error. I'm not a coder or machine learning specialist, so I've been following the provided instructions as closely as possible. However, despite numerous attempts and modifications, the error persists.

When executing the training command, I receive the following error:

`RuntimeError: cannot reshape tensor of 0 elements into shape [0, -1] because the unspecified dimension size -1 can be any value and is ambiguous`

The error seems to occur during the initialization of the model architecture, particularly at a step involving weight normalization.

Steps Taken:

- Verified the dataset path and ensured the preprocessed dataset is correctly located.
- Set the model name, architecture, regularization, and other parameters explicitly.
- Checked that the Google Drive is properly mounted in Colab and the paths are accessible.
- Ensured no syntax errors in the notebook cells, and all cells are executed in the correct order.
- Attempted to update the RAVE package and its dependencies to the latest versions.

Despite these steps, the error occurs consistently. Below is the training command used:

`!/content/miniconda/bin/rave train --config /content/miniconda/lib/python3.9/site-packages/rave/configs/v2.gin --config /content/miniconda/lib/python3.9/site-packages/rave/configs/wasserstein.gin --db_path "/content/dataset" --name "turbom" --val_every 2500`

Could you please provide some insight into what might be causing this issue? Any guidance on resolving this would be greatly appreciated, as I am at an impasse and unable to proceed with training the model.

Thank you for your time and assistance!

**Comment by ctralie** [comment]
This seems like a related issue:
https://github.com/acids-ircam/RAVE/issues/273

I had a similar issue, and it was fixed when I explicitly specified --channels 1 in my training
